hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java

```diff
@@ -190,6 +193,28 @@ public boolean checkAndEnforceThrottle(
         return false;
     }
 
+    /**
+     * Checks if capacity has been breached in the ops duration throttle.
+     *
+     * @param now the instant of consensus time the transaction throttling should be checked for
+     * @return whether the transaction should be throttled
+     */
+    public boolean checkAndEnforceOpsDurationThrottle(final long currentOpsDuration, @NonNull final Instant now) {
+        if (throttleType == NOOP_THROTTLE) {
+            return false;
+        }
+        contractOpsDurationThrottle.resetLastAllowedUse();
+
+        final boolean shouldThrottleByOpsDuration =
```

Comment:
Should this be looked up once and remembered at the start of the EVM transaction?

Reply:
will address in follow up pr